File: learn_py_with_js/os/listDir.mjs
import fs from 'fs'
import path from 'path'
import {fileURLToPath} from 'node:url';
import {dirname} from 'node:path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);


function listItems(directoryPath) {
    const items = fs.readdirSync(directoryPath)
    items.forEach(item => {
        const itemPath = path.join(directoryPath, item)
        const stats = fs.statSync(itemPath)
        console.log(`${item} is a ${stats.isDirectory() ? 'directory' : 'file'}`)
    })
}

// listItems(path.join(__dirname))

listItems('../')

File: learn_py_with_js/os/list_dir.py
import os


def list_items(directory):
    items = os.listdir(directory)

    for item in items:
        item_path = os.path.join(directory, item)
        if os.path.isdir(item_path):
            print(f"{item} is a directory")
        else:
            print(f"{item} is a file")

list_items('../')